RecoLocalCalo/EcalRecProducers/plugins/alpaka/EcalUncalibRecHitPhase2WeightsAlgoPortable.dev.cc

```diff
@@ -0,0 +1,111 @@
+#include "DataFormats/EcalDigi/interface/EcalDataFrame_Ph2.h"
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+
+#include "EcalUncalibRecHitPhase2WeightsAlgoPortable.h"
+
+#include "DataFormats/EcalDigi/interface/alpaka/EcalDigiPhase2DeviceCollection.h"
+#include "DataFormats/EcalRecHit/interface/alpaka/EcalUncalibratedRecHitDeviceCollection.h"
+#include <alpaka/alpaka.hpp>
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/traits.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+
+#include "DataFormats/EcalRecHit/interface/EcalUncalibratedRecHit.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  namespace ecal {
+    namespace weights {
+
+      using namespace cms::alpakatools;
+
+      class Phase2WeightsKernel {
+      public:
+        template <typename TAcc>
+        ALPAKA_FN_ACC void operator()(TAcc const &acc,
+                                      double const *weightsdata,
+                                      double const *timeWeightsdata,
+                                      EcalDigiPhase2DeviceCollection::ConstView digisDev,
+                                      EcalUncalibratedRecHitDeviceCollection::View uncalibratedRecHitsDev) const;
+      };
+
+      template <typename TAcc>
+      ALPAKA_FN_ACC void Phase2WeightsKernel::operator()(
+          TAcc const &acc,
+          double const *weightsData,
+          double const *timeWeightsdata,
+          EcalDigiPhase2DeviceCollection::ConstView digisDev,
+          EcalUncalibratedRecHitDeviceCollection::View uncalibratedRecHitsDev) const {
+        constexpr int nsamples = EcalDataFrame_Ph2::MAXSAMPLES;
```

Comment:
I'd suggest to use either `ecalPh2::sampleSize` or `EcalDataFrame_Ph2::MAXSAMPLES` (I see the latter is defined to be the same as the former) both here and where `weightsData` and `timeWeightsdata` are allocated/defined so that a future reader can easily convince themselves that the loop below does not go over the bounds of `weightsData`/`timeWeightsdata`.